.github/workflows/e2e.yml

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "End to End Tests"
+on:
+  push:
+    # run on pushes to master or release/*
+    branches:
+      - master
+      - release/*
+  pull_request:
+    # run on pull requests targeting master
+    branches:
+      - master
+jobs:
+  end-to-end:
+    runs-on: [self-hosted, e2e-worker]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Clean Project State
+      run: |
+        # clean GCP resources
+        .github/workflows/e2e_scripts/clean.sh
```

Comment:
can we reuse destroy.sh?

Reply:
No, destroy.sh is simple because it wipes the entire project. We need to keep the project, but individually delete resources